### PR TITLE
TOML docs: release (vars) config

### PR DESCRIPTION
**Pull request by chrzaszcz**
- Reorder the configuration docs, now the first file is the really starting page.
- Rename 'basic configuration' (which was not describing the basic usage) to 'release options'.
- Update the release options (`vars-toml.config`) with the TOML format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[8585.1](https://travis-ci.org/esl/MongooseIM/jobs/731236966) / Erlang 23.0.3 / small_tests / 571668c0c546c72367833d7e0cda6830e4b8644c
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2890/8585/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2890/8585/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2020-09-29_10.32.32/lib.mongooseim.logs/run.2020-09-29_10.32.32/suite.log.html)

File: doc/advanced-configuration/release-options.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+* **Type:** parameter
+* **Option:** [`general.hosts`](../../advanced-configuration/general#generalhosts)
+* **Syntax:** comma-separated list of strings
+* **Example:** `{hosts, "\"localhost\", \"domain2\""}.`
+
+## host_config
+
+* **Type:** block
+* **Option:** [`host_config`](../../advanced-configuration/host_config.md)
```
Comment (janciesla8818) [inline doc/advanced-configuration/release-options.md:50]:
```suggestion
* **Option:** [`host_config`](../../advanced-configuration/host_config)
```
I've run it locally and was not working. I think we add `.md` for absolute paths.

File: doc/advanced-configuration/release-options.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+  [host_config.auth]
+    methods = [\"anonymous\"]
+"}.
+```
+
+## auth_ldap
+
+* **Type:** block
+* **Option:** [`auth.ldap`](../../authentication-methods/ldap.md)
```
Comment (janciesla8818) [inline doc/advanced-configuration/release-options.md:67]:
```suggestion
* **Option:** [`auth.ldap`](../../authentication-methods/ldap)
```

File: doc/advanced-configuration/release-options.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+```
+
+## s2s_default_policy
+
+* **Type:** parameter
+* **Option:** [`s2s.default_policy`](../../advanced-configuration/s2s#s2sdefault_policy)
+* **Syntax:** string
+* **Example:** `{s2s_default_policy, "\"deny\""}.`
+
+## outgoing_s2s_port
```
Comment (janciesla8818) [inline doc/advanced-configuration/release-options.md:109]:
```suggestion
## s2s_outgoing_port
```

File: doc/advanced-configuration/release-options.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+* **Example:** `{c2s_port, "5222"}.`
+
+## s2s_port
+
+* **Type:** parameter
+* **Option:** [`listen.s2s.port`](../../advanced-configuration/listen#listenport)
+* **Syntax:** integer
+* **Example:** `{s2s_port, "5269"}.`
+
+## cowboy_port
```
Comment (janciesla8818) [inline doc/advanced-configuration/release-options.md:130]: Do we keep `cowboy` here or should it be `http_port`?

File: doc/user-guide/client-certificate.md [hunk 5]
```diff
@@ -44,7 +44,7 @@ For the details please refer to [XEP-0178 Best Practices for Use of SASL EXTERNA
 You need to enable one of the following authentication backends by using the [`auth_method` option](../Advanced-configuration.md#authentication) in the MongooseIM configuration file.
```
Comment (janciesla8818) [inline doc/user-guide/client-certificate.md:44]:
```suggestion
You need to enable one of the following authentication methods by using the [`auth_method` option](../Advanced-configuration.md#authentication) in the MongooseIM configuration file.
```

Comment (janciesla8818) [reply doc/user-guide/client-certificate.md:44]: Also, please change `ingore` to `ignore` in line 169, I couldn't comment there.

File: doc/user-guide/client-certificate.md [hunk 6]
```diff
@@ -44,7 +44,7 @@ For the details please refer to [XEP-0178 Best Practices for Use of SASL EXTERNA
...
 
 * `pki` - accepts user credentials,
-* `http` - accepts user credentials if the provided certificate is [known and valid](../authentication-backends/HTTP-authentication-module.md#method-get_certs)
+* `http` - accepts user credentials if the provided certificate is [known and valid](../authentication-backends/http#method-get_certs)
```
Comment (janciesla8818) [inline doc/user-guide/client-certificate.md:47]:
```suggestion
* `http` - accepts user credentials if the provided certificate is [known and valid](../../authentication-methods/http#method-get_certs)
```

**Review by janciesla8818 — COMMENTED** [review]
Thanks for taking care of this so quickly 💪 

I only have few minor comments.

File: doc/advanced-configuration/release-options.md (hunk 3 above)
Comment (chrzaszcz) [reply doc/advanced-configuration/release-options.md:109]: The other way around, but a mistake anyway, good catch!

File: doc/advanced-configuration/release-options.md (hunk 4 above)
Comment (chrzaszcz) [reply doc/advanced-configuration/release-options.md:130]: I wouldn't want to change this as we use it in the template variables.

File: doc/user-guide/client-certificate.md (hunk 5 above)
Comment (chrzaszcz) [reply doc/user-guide/client-certificate.md:44]: I did nothing to this file yet, but I will fix this.

File: doc/user-guide/client-certificate.md (hunk 6 above)
Comment (chrzaszcz) [reply doc/user-guide/client-certificate.md:47]: I did nothing to this file yet (it's a separate story), but I will fix this.

**Comment by chrzaszcz** [comment]
Thanks @janciesla8818, it's interesting that `mkdocs` does not complain about the`md` suffixes and complains about some other cases... it turns out you need to try to follow the link manually to verify it.

File: doc/user-guide/client-certificate.md (hunk 6 above)
Comment (janciesla8818) [reply doc/user-guide/client-certificate.md]: I thought that you changed the `HTTP-authentication-module` to `http`. Anyway, thanks for changing that.

**Review by janciesla8818 — APPROVED** [review]

**Review by NelsonVides — APPROVED** [review]

**Pull request merged**